### ISSUE TITLE
[Bug]: Connection Refused Error while running OpenHands

### ISSUE THREAD

**Issue by AhmedProab** [All-Hands-AI/OpenHands#5427]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

I encountered a bug while trying to run OpenHands. When attempting to run the application, I received the following error message: 

Request failed with status code 500
HTTPConnectionPool(host='host.docker.internal', port=33817): Max retries exceeded with url: /execute_action (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xffff644c1460>: Failed to establish a new connection: [Errno 111] Connection refused'))

Steps to reproduce the bug:
Start the OpenHands project.
Attempt to connect to the application in the browser.
The error Connection refused occurs, and the app does not load properly.
OpenHands Installation:

### OpenHands Installation

Docker command in README

### OpenHands Version

0.15.0 - 2025-12-04

### Operating System

MacOS

### Logs, Errors, Screenshots, and Additional Context

- Restarting the machine did not resolve the issue.
- The agent is continuously showing 'Agent is running task' but no results are produced.
- The agent sometimes shows 'Agent is stuck.'
- Screenshots are attached for further context.

<img width="811" alt="Screenshot 2024-12-05 at 8 56 55 PM" src="https://github.com/user-attachments/assets/95fd5e14-cc24-4729-8553-7ce0b66d4f4c">
<img width="423" alt="Screenshot 2024-12-05 at 8 54 30 PM" src="https://github.com/user-attachments/assets/c5c456ca-05c7-4344-b0db-a16e2a5ee535">

**Comment by mamoodi** [comment]
Can't reproduce this on mac.
Are you running the docker command exactly as the README?
```
docker run -it --pull=always \
    -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:0.15-nikolaik \
    -e LOG_ALL_EVENTS=true \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -p 3000:3000 \
    --add-host host.docker.internal:host-gateway \
    --name openhands-app \
    docker.all-hands.dev/all-hands-ai/openhands:0.15
```

Did you have previous versions working?

**Comment by AhmedProab** [comment]
Thank you so much for your assistance!

Yes, I am using Docker on my desktop. To resolve the issue, I deleted everything and followed the steps outlined in the [official documentation](https://docs.all-hands.dev/modules/usage/installation). I reinstalled OpenHands using the provided commands:

bash
Copy code
docker pull docker.all-hands.dev/all-hands-ai/runtime:0.15-nikolaik

docker run -it --rm --pull=always \
    -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:0.15-nikolaik \
    -e LOG_ALL_EVENTS=true \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -p 3000:3000 \
    --add-host host.docker.internal:host-gateway \
    --name openhands-app \
    docker.all-hands.dev/all-hands-ai/openhands:0.15
Thankfully, it's working perfectly now. Thanks again for your kind support! 😊
